uniffi_bindgen/src/templates/scaffolding_template.rs

```diff
@@ -2,6 +2,12 @@
 // Trust me, you don't want to mess with it!
 {% import "macros.rs" as rs %}
 
+// Check for compatibility between `uniffi` and `uniffi_bindgen` versions.
+// Note that we have an error message on the same line as the assertion.
+// This is important, because if the assertion fails, the compiler only
+// seems to show that single line as context for the user.
+uniffi::assert_compatible_version!("{{ uniffi_version }}"); // Please check that you depend on version {{ uniffi_version }} of the `uniffi` crate.
```

Comment:
To check my understanding: this is about checking that the `uniffi_bindgen`-generated scaffolding is being compiled with the same version of `uniffi`-runtime, and that the later compiled when the generated scaffold and runtime is being compiled. Is this correct?

Reply:
> and that the later compiled when the generated scaffold and runtime is being compiled. 

I don't fully parse this part TBH, but what you're saying here otherwise sounds right to me. If I were to try to round-trip that you're saying here through my own parser and renderer it would come out as:

"""
this is about checking that the `uniffi_bindgen`-generated scaffolding is being compiled against the same version of the `uniffi` runtime crate as was used when the `uniffi_bindgen` executable itself was compiled.
"""

Or put another way, if `uniffi-bindgen --version` reports `v1.2.3`, then any bindings it generates will generate a compile-time error if they're being compiled against a version of `uniffi` other than `v1.2.3`.

Reply:
(I'll also note that this problem would go away entirely if we switched to using macros rather than an external bindgen command, but that's something to discuss for the future)